### ISSUE TITLE
Mark `callMethod` as `discardable`

### ISSUE THREAD

**Issue by LemonBoy** [yglukhov/nimpy#30]
In order to avoid a slew of useless `discard` I'd suggest to mark the `callMethod` returning a `PyObject` as discardable.

**Comment by yglukhov** [comment]
I'd really try to avoid `discardable` as long as possible and would welcome other solutions, like passing ret typedesc to the function. I'd like to here more complaints before adding `discardable` to the public interface about which I will regret after some time but will not be able to revert it without breaking.

**Comment by LemonBoy** [comment]
Hmm, what I really meant was to have `.()` as discardable since that'd be a great improvement from an ergonomic point of view.

I wholeheartedly agree that putting `discardable` is a code smell.

**Comment by yglukhov** [comment]
> what I really meant was...

Yes, that's precisely how I understood it. But is this a real problem though? Why would anyone want to call lots of python from nim?

**Comment by tim-st** [comment]
I implement some python code using `nimpy` e.g. https://github.com/explosion/spaCy to support and test my nim code. Also other modules which are not available in nim I call from python.
I think it's not a real problem but would be nice to have.

**Comment by LemonBoy** [comment]
> Why would anyone want to call lots of python from nim

Because of Matplotlib and Seaborn :)

**Comment by slangmgh** [comment]
I hope the it is ``discardable``, so the code in nim is just as same as in python.
```nim
proc plot_data(data: HistData) =
   if not init_python(): return

   plt.rcParams["font.sans-serif"]=["SimHei"]
   plt.rcParams["axes.unicode_minus"] = false

   plt.title(data.code)
   let vtime = map(0..<data.len, i -> data.get_small_time(i))
   set_xaxis(vtime, 8)
   let fig = plt.figure(data.code)
   fig.autofmt_xdate()
   plt.plot(data.vclose)
```
